data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/extension/KinesisLeaseCoordinationTableConfig.java

```diff
@@ -0,0 +1,22 @@
+package org.opensearch.dataprepper.plugins.kinesis.extension;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+import lombok.NonNull;
+import software.amazon.awssdk.regions.Region;
+
+@Getter
+public class KinesisLeaseCoordinationTableConfig {
+
+    @JsonProperty("table_name")
+    @NonNull
+    private String tableName;
```

Comment:
How will this work if the user has multiple streams?

Reply:
The partition key for this table is `LEASE_KEY` which will be a combination of the stream identifier and the shardId. Like `<streamIdentifier>:<shardId>`. For e.g. `<accountId>:<streamName>:<streamCreationEpochTimestamp>:<shardId-xxxxx>` where the `streamIdentifier` is a combination of `<accountId>:<streamName>:<streamCreationEpochTimestamp>`.